### ISSUE TITLE
Web: Build226 TypeError is not a function when calling a function with default arguments

### ISSUE THREAD

**Issue by hfhbd** [JetBrains/compose-jb#811]
Hey, 
Trying out the unreleased build226 I am getting a: `TypeError: this._$attrs_9._v is not a function` if I use a @Composable function with an empty default lambda in the main sourceSet. If I copy the same function to the test sourceSet, the test succeeds. 

Bug as project: 
[bootstrap-compose Kopie.zip](https://github.com/JetBrains/compose-jb/files/6704654/bootstrap-compose.Kopie.zip)

```kotlin
// main sourceSet
@Composable
public fun ButtonMain(
    title: String,
    color: Color = Color.Primary,
    type: ButtonType = ButtonType.Submit,
    attrs: AttrsBuilder<HTMLButtonElement>.() -> Unit = { },
    action: () -> Unit
) {
  Button(attrs = {
    classes("btn", "btn-$color")
    attrs()
    type(type)
    onClick {
        action()
    }
}) {
    Text(title)
    }
}

// test sourceSet
@Composable
public fun ButtonTest(     // same function, just copied
    title: String,
    color: Color = Color.Primary,
    type: ButtonType = ButtonType.Submit,
    attrs: AttrsBuilder<HTMLButtonElement>.() -> Unit = { },
    action: () -> Unit
) {
    Button(attrs = {
        classes("btn", "btn-$color")
        attrs()
        type(type)
        onClick {
            action()
        }
    }) {
        Text(title)
    }
}

@Test
fun variableButtonTestModule() = runTest {
    val s: @Composable () -> Unit = @Composable { ButtonTest("demo") { } }
    composition {
        s()
    }
    assertEquals("""<button type="submit" class="btn btn-primary">demo</button>""", root.innerHTML) // no errors
}

@Test
fun variableButtonMainModule() = runTest {
    val s: @Composable () -> Unit = @Composable { ButtonMain("demo") { } }
    composition {
        s()
    }
    assertEquals("""<button type="submit" class="btn btn-primary">demo</button>""", root.innerHTML) // TypeError: this._$attrs_9._v is not a function
}
```

**Comment by eymar** [comment]
thanks for reporting!
the first version of the fix is ready (but not merged and release yet) - https://github.com/JetBrains/androidx/commit/398f93c9ec454c8a75cfb48e9764c82e400afa4e
If the fix is okayish for others, then we can expect the issue will be solved by the next release (not earlier than build227)

**Comment by hfhbd** [comment]
The temporary workaround would be to use nullables instead:
```kotlin
@Composable
fun Something(handler: (() -> Unit)? = null) { ... }
```

**Comment by hfhbd** [comment]
@eymar This error is not limited to a lambda parameter, but every default parameter is `null` at runtime, which results into a `TypeError`.
Eg:
```kotlin
@Composable
public fun Button(
    title: String,
    color: Color = Color.Primary,
    type: ButtonType = ButtonType.Submit,
    attrs: AttrBuilderContext<HTMLButtonElement>? = null,
    action: () -> Unit
) {
    Button(attrs = {
        classes("btn", "btn-$color")
        attrs?.invoke(this)
        type(type)
        onClick {
            action()
        }
    }) {
        Text(title)
    }
}
```
Failing with: `TypeError: Cannot read property '_str' of null` where `_str` is `ButtonType.Submit.str`, but `type` is `null` at runtime.

**Comment by eymar** [comment]
closing this issue: it was fixed in 0.5.0-build227

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6404">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
